src/algebra/group/conj.lean

```diff
@@ -46,9 +49,9 @@ protected lemma monoid_hom.map_is_conj (f : α →* β) {a b : α} : is_conj a b
 end monoid
 
 section cancel_monoid
-variables [cancel_monoid α]
--- These lemmas hold for either `left_cancel_monoid` or `right_cancel_monoid`,
--- with slightly different proofs; so far these don't seem necessary.
+
+variables [right_cancel_monoid α]
+-- These lemmas also hold for `left_cancel_monoid` - so far this is not necessary.
```

Comment:
Right, but it makes the statement assymmetric; let me try and mind the comment I made when this was previously suggested

Reply:
Hmm, I guess I didn't say much more than that: https://github.com/leanprover-community/mathlib/pull/13109#discussion_r841057966

My fear is that "generalizing" in this kind of case leads to a pile of lemmas that only work for left cancellation, and a pile that only work for right. I'd rather leave the symmetric version to avoid this split, and leave a comment as a "we're missing a better generalization" marker.

Reply:
I ungeneralised. I think that we need a better approach to dealing with these things. Can we do an order-dual like thing here? Create the monoid with a *M b = b * a?

Reply:
It sounds like you're asking for `mul_opposite`, but I'm not sure how that helps

Reply:
This theorem on the mul_opposite should prove the left cancel version, right?

Reply:
Sure; but then there's the problem that we have two lemmas with exactly the same statement yet different typeclass assumptions, and we have to somehow work an extra left or right into a lemma that already has a left or right in its name

Reply:
What else can we do? Do we need a "can somehow cancel" typeclaas? That seems weird though